### PR TITLE
Implement "--inverted" flag

### LINKED ISSUE #22: Icicle graphs
The [`--inverted` flag](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L149), apart from [changing the title of the plot](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L205), changes the flame graph such that the entire plot is [up-side-down](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L1067L1068) (note that this also affects [the JavaScript](https://github.com/brendangregg/FlameGraph/blob/f857ebc94bfe2a9bfdc4f1536ebacfb7466f69ba/flamegraph.pl#L864)). We should probably support that flag too.

### PR DESCRIPTION
**Pull request by busarovalex**
Closes #22

### CODE REVIEW

File: src/flamegraph/mod.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ use quick_xml::{
     events::{BytesEnd, BytesStart, BytesText, Event},
     Writer,
 };
+use std::default::Default;
```
Comment (jonhoo) [inline src/flamegraph/mod.rs]: You don't need this `use`; `Default` is in the Rust prelude :)

File: src/flamegraph/mod.rs [hunk 2]
```diff
@@ -305,3 +324,18 @@ fn deannotate(f: &str) -> &str {
     }
     f
 }
+
+impl Default for Direction {
```
Comment (jonhoo) [inline src/flamegraph/mod.rs]: Do we even need an `impl Default for Direction`? It doesn't seem to be used anywhere?

File: src/flamegraph/mod.rs [hunk 3]
```diff
@@ -305,3 +324,18 @@ fn deannotate(f: &str) -> &str {
...
+    fn default() -> Self {
+        Direction::Straight
+    }
+}
+
+impl Direction {
```
Comment (jonhoo) [inline src/flamegraph/mod.rs]: I don't think `is_inverted` is used anywhere?

File: src/flamegraph/svg.rs [hunk 4]
```diff
@@ -140,7 +146,7 @@ var searchcolor = 'rgb(230,0,230)';",
             size: super::FONTSIZE + 5,
             x: (super::IMAGEWIDTH / 2) as f64,
             y: (super::FONTSIZE * 2) as f64,
-            text: "Flame Graph".into(),
+            text: opt.title.clone().into(),
```
Comment (jonhoo) [inline src/flamegraph/svg.rs]: I don't think you need the `clone()` here? Specifically, I think `&str` should work just fine:
```rust
(&*opt.title).into()
```

**Comment by jonhoo** [comment]
Looks good to me once the few comments above have been sorted out! Thanks :D

File: src/flamegraph/mod.rs (hunk 2 above)
Comment (jasonrhansen) [reply src/flamegraph/mod.rs]: I think we need Direction to implement Default since Options does.

Comment (jonhoo) [reply src/flamegraph/mod.rs]: Given that we [directly construct `Options`](https://github.com/jonhoo/inferno/pull/48/files#diff-5b9f024b0146520495e6e19fa806f9daR47), why does `Options` have to `impl Default` then? :p

Comment (jasonrhansen) [reply src/flamegraph/mod.rs]: What about when someone wants to use it as a library? Wouldn't we want the ability to just use the default options?

Comment (jonhoo) [reply src/flamegraph/mod.rs]: Ah, hmm, I suppose that's true... In fact, arguably what we should do in `from_args` is construct a `default` and then only override with what the args say...

File: src/flamegraph/mod.rs (hunk 1 above)
Comment (busarovalex) [reply src/flamegraph/mod.rs]: Done

File: src/flamegraph/mod.rs (hunk 3 above)
Comment (busarovalex) [reply src/flamegraph/mod.rs]: Removed this method, added Copy,Clone,Eq,PartialEq to Direction.
In `svg.rs` used `opt.direction == Direction::Inverted`

File: src/flamegraph/svg.rs (hunk 4 above)
Comment (busarovalex) [reply src/flamegraph/svg.rs]: Done

File: src/bin/flamegraph.rs [hunk 5]
```diff
@@ -14,6 +14,12 @@ struct Opt {
     #[structopt(long = "nameattr")]
     nameattr_file: Option<PathBuf>,
 
+    /// The `--inverted` flag.
+    /// Changes the flame graph such that the entire plot is up-side-down.
+    /// Also, changing the title of the plot.
```
Comment (jonhoo) [inline src/bin/flamegraph.rs]: Actually, sorry, one more thing. Let's make this a little less noisy, and make it just:
```rust
/// Plot the flame graph up-side-down.
```

Comment (busarovalex) [reply src/bin/flamegraph.rs]: Done

**Comment by jonhoo** [comment]
Great, thank you!

**Pull request merged**